ExampleMod/Content/Buffs/AnimatedBuff.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Buffs
+{
+	//This file contains two classes, the ModSystem responsible for updating the animation, and the ModBuff itself.
+
+	//If you are not interested in making your buff animated, you don't need this ModSystem class here.
+	public class AnimatedBuffSystem : ModSystem
+	{
+		public const int FrameCount = 4; //Amount of frames we have on our animation spritesheet.
+		public const int AnimationSpeed = 60; //In ticks.
+
+		//This is a property, in this case it allows us to restrict the setter to only this class, meaning that "AnimatedBuffSystem.Frame = 5" in another class would not let us compile.
+		public static int Frame { get; private set; } = 0;
+
+		private static int FrameCounter { get; set; } = 0;
+
+		public override void PreUpdateEntities() {
+			//This hook runs once per game tick, in the same place where other animation updates take place.
+			//If we were to put this code in AnimatedBuff.PreDraw, even though vanilla makes sure to only have one buff per player,
+			//it's possible that another mod might invoke drawing of buffs, which will then speed up the animation due to code running multiple times per tick.
+			if (!Main.dedServ && (Main.hasFocus || Main.netMode != NetmodeID.SinglePlayer)) {
+				//The above conditions are the same vanilla uses to update animations.
+
+				FrameCounter++;
+				if (FrameCounter > AnimationSpeed) {
+					FrameCounter = 0;
+					Frame++;
+					if (Frame >= FrameCount) {
+						Frame = 0;
+					}
+				}
+			}
+		}
+	}
+
+	//This buff has an extra animation spritesheet, and also showcases PreDraw specifically.
+	//(We keep the autoloaded texture as one frame in case other mods need to access the buff sprite directly and aren't aware of it having special draw code).
+	public class AnimatedBuff : ModBuff
+	{
+		private Asset<Texture2D> animatedTexture;
+
+		public override void SetDefaults() {
+			DisplayName.SetDefault("Animated Buff");
+			Description.SetDefault("Animates.");
+
+			if (Main.netMode != NetmodeID.Server) {
+				//Do NOT load textures on the server!
+				animatedTexture = Mod.GetTexture("Content/Buffs/AnimatedBuff_Animation");
+			}
+		}
+
+		public override void Unload() {
+			animatedTexture = null;
+		}
+
+		public override bool PreDraw(SpriteBatch spriteBatch, int buffIndex, ref BuffDrawParams drawParams) {
+			//We draw our special texture here, the animation takes place in AnimatedBuffSystem.
+
+			//Use our animation spritesheet.
+			Texture2D ourTexture = animatedTexture.Value;
+			//Choose the frame to display.
+			//We reference the frame count and frame from the ModSystem here.
+			Rectangle ourSourceRectangle = ourTexture.Frame(verticalFrames: AnimatedBuffSystem.FrameCount, frameY: AnimatedBuffSystem.Frame);
+
```

Comment:
There indeed is zero use in that system, lol.
```suggestion
	//This buff has an extra animation spritesheet, and also showcases PreDraw specifically.
	//(We keep the autoloaded texture as one frame in case other mods need to access the buff sprite directly and aren't aware of it having special draw code).
	public class AnimatedBuff : ModBuff
	{
		public const int FrameCount = 4; //Amount of frames we have on our animation spritesheet.
		public const int AnimationSpeed = 60; //In ticks.
		
		private static Asset<Texture2D> animatedTexture;

		public override void SetDefaults() {
			DisplayName.SetDefault("Animated Buff");
			Description.SetDefault("Animates.");

			if (Main.netMode != NetmodeID.Server) {
				//Do NOT load textures on the server!
				animatedTexture = Mod.GetTexture("Content/Buffs/AnimatedBuff_Animation");
			}
		}

		public override void Unload() {
			animatedTexture = null;
		}

		public override bool PreDraw(SpriteBatch spriteBatch, int buffIndex, ref BuffDrawParams drawParams) {
			//We draw our special texture here with a specific animation.

			//Use our animation spritesheet.
			Texture2D ourTexture = animatedTexture.Value;
			//Choose the frame to display, based on constants and the game's tick counts.
			Rectangle ourSourceRectangle = ourTexture.Frame(verticalFrames: FrameCount, frameY: Main.GameUpdateCount / AnimationSpeed % FrameCount);

```

Reply:
Overengineering at its finest :cowboy_hat_face: